flaml/autogen/agent/user_proxy_agent.py

```diff
@@ -90,23 +96,109 @@ def _execute_code(self, code_blocks):
                 return exitcode, logs_all
         return exitcode, logs_all
 
-    def auto_reply(self, message, sender, default_reply=""):
+    @staticmethod
+    def _format_json_str(jstr):
+        """Remove newlines outside of quotes, and hanlde JSON escape sequences.
+
+        1. this function removes the newline in the query outside of quotes otherwise json.loads(s) will fail.
+            Ex 1:
+            "{\n"tool": "python",\n"query": "print('hello')\nprint('world')"\n}" -> "{"tool": "python","query": "print('hello')\nprint('world')"}"
+            Ex 2:
+            "{\n  \"location\": \"Boston, MA\"\n}" -> "{"location": "Boston, MA"}"
+
+        2. this function also handles JSON escape sequences inside quotes,
+            Ex 1:
+            '{"args": "a\na\na\ta"}' -> '{"args": "a\\na\\na\\ta"}'
+        """
+        result = []
+        inside_quotes = False
+        last_char = " "
+        for char in jstr:
+            if last_char != "\\" and char == '"':
+                inside_quotes = not inside_quotes
+            last_char = char
+            if not inside_quotes and char == "\n":
+                continue
+            if inside_quotes and char == "\n":
+                char = "\\n"
+            if inside_quotes and char == "\t":
+                char = "\\t"
+            result.append(char)
+        return "".join(result)
+
+    def _extract_args(self, input_string: str):
+        """Extract arguments from a json-like string and put it into a dict.
+
+        Args:
+            input_string: string to extract arguments from.
+
+        Returns:
+            a dictionary or None.
+        """
+        input_string = self._format_json_str(input_string)
+        try:
+            args = json.loads(input_string)
+            return args
+        except json.JSONDecodeError:
+            return {}
```

Comment:
catching this error here would prevent the feedback about the json decode error to be sent back to the assistant agent.